hazelcast/src/main/java/com/hazelcast/nio/tcp/SingleProtocolDecoder.java

```diff
@@ -97,6 +100,13 @@ private String loadProtocol() {
         return bytesToString(protocolBytes);
     }
 
+    private void initConnection() {
+        if (supportedProtocol == ProtocolType.MEMBER) {
+            TcpIpConnection connection = (TcpIpConnection) channel.attributeMap().get(TcpIpConnection.class);
+            connection.setType(ConnectionType.MEMBER);
```

Comment:
This is the fix to ensure connection type is set immediately after protocol bytes have been received -> it will be available in `BindHandler` processing